### ISSUE TITLE
Folder persist after being successfully deleted in blob container

### ISSUE THREAD

**Issue by kediwu0331** [microsoft/AzureStorageExplorer#2665]
**Storage Explorer Version**: 1.12.0
**Build Number**: 20200116.10
**Platform/OS**: macOS Catalina
**Architecture**: x86
**Regression From**: 

## Bug Description ##
After I deleted the 'folder', it still persists in the explorer UI. And when I tried to delete again, I saw this error: {
  "message": "\"failed to perform remove command due to error: nothing found to remove\""
}

## Steps to Reproduce ##
1. delete a folder in a blob container

## Expected Experience ##
The folder will persist. And when you delete again, there will be an error.

## Actual Experience ##
![image](https://user-images.githubusercontent.com/31940276/75184969-245b5700-56fa-11ea-8275-0a126fa06f3e.png)

**Comment by JasonYeMSFT** [comment]
Can you try the failed delete again and share the AzCopy log file with us? You should be able to find the log file in your ~/.azcopy directory. Also what about the remaining blobs in the problematic directory? Can you share their path with us so we can see if AzCopy is handling the paths correctly?

**Comment by kediwu0331** [comment]
Hi Jason,

The full path is journey-test/adobecom/model_instance/20200112-fatigue-norecency/. Here is what happened in the folder: the deepest folder structure was journey-test/adobecom/model_instance/20200112-fatigue-norecency/email/fatigue/A/training_data/*. After the deletion, the files under training data are gone. But the training_data folder still persists, showing it is a 0B block blob, which is weird.

Here is the AzCopy Command. Sorry I was not able to find the log. Don't see ~/.azcopy.

export AZCOPY_CRED_TYPE=Anonymous;
export AZCOPY_USER_AGENT_PREFIX=Microsoft Azure Storage Explorer, 1.12.0, darwin, ;
./azcopy remove "https://adobecomjourneyaiintstrg.blob.core.windows.net/journey-test/adobecom/model_instance/20200112-fatigue-norecency/?se=2020-03-27T19%3A14%3A53Z&sp=rdl&sv=2018-03-28&sr=c&sig=vZ8E7H0o6QyQEzIQGCpZYS0AG107lhFJUu1htwX0YC4%3D" --recursive
unset AZCOPY_CRED_TYPE;
unset AZCOPY_USER_AGENT_PREFIX;

**Comment by JasonYeMSFT** [comment]
@kediwu0331 we have investigated the issue and found that the problem was caused by AzCopy intentionally skipping the blobs with metadata `hdi_isFolder=True`. The metadata is used for representing empty directories due to the fact that Azure Blob Storage doesn't natively support directories. For now, you can go to portal and use the blob viewer there to remove those blobs.

**Comment by JasonYeMSFT** [comment]
@kediwu0331 I just realized that I may have accidentally deleted some files in your journey-test container 😥. Super sorry about that. Are you able to regenerate the data? Please let us know if there is anything we can do to recover the data.

**Comment by kediwu0331** [comment]
Hi Jason, could you let me know which files/folders you deleted? Thanks

**Comment by JasonYeMSFT** [comment]
I selected and deleted the journey-test/adobecom (blob) and see about 601 files getting removed, which due to the same bug, AzCopy goes to delete the files in the adobecom folder. I have the AzCopy log available so I can parse the log to give you a detailed list of files.

**Comment by kediwu0331** [comment]
Hi Jason, yeah please share the list of files. Thanks!

**Comment by JasonYeMSFT** [comment]
Here is the list of blobs that I found in the AzCopy log. Each line is the confirmation log message for deleting a single blob item.

[accidentally_removed_blobs.txt](https://github.com/microsoft/AzureStorageExplorer/files/4264712/accidentally_removed_blobs.txt)

**Comment by JasonYeMSFT** [comment]
Here is a condensed list of files where each line is exactly the relative path for the blob under `journey-test/adobecom`
[condensed_list_of_files.txt](https://github.com/microsoft/AzureStorageExplorer/files/4264721/condensed_list_of_files.txt)

**Comment by kediwu0331** [comment]
Hi Jason, that's fine. Don't worry about it. The files you deleted are our test models. But for this storage explorer bug, when can it be fixed?

**Comment by JasonYeMSFT** [comment]
Since we are going to release 1.13.0 mid march and there is no room for us to fit this in it, we are going to push it and relase the fix in 1.14.0.

**Comment by MRayermannMSFT** [comment]
Let's push the AzCopy team to get this fixed in time for 1.14

**Comment by MRayermannMSFT** [comment]
For the record, anyone blocked by this can always:
- call the Blob APIs yourself to delete the blobs
- try to use whatever you originally used to create the blobs to delete them
- try removing the `hdi_isfolder=true` metadata from the blobs (right click -> properties) and then deleting (for this solution, we're not 100% sure what the effects of deleting these blobs will be on the tool that was used to create the blobs, we strongly recommend that you make sure doing this is safe to do in the context of your tool)

**Comment by MRayermannMSFT** [comment]
Duplicate of https://github.com/microsoft/AzureStorageExplorer/issues/2466
